pkg/catalog/generator.go

```diff
@@ -130,6 +131,21 @@ func handleSourceURL(importPath string) (string, error) {
 	return buildLink(remote, importPath), nil
 }
 
+// NewProjectFromJson generates a generator object with a json byte input (of a sysl module) instead of a sysl module
+func NewProjectFromJson(
```

Comment:
I wonder if this could be implemented instead as an additional method on generator, that way you wouldn't have to handle a lot of extra stuff that's already handled by NewProject e.g
```go
func (p *Generator) FromJSON(module []byte) *Generator {
	if err := json.Unmarshal(module, p.RootModule); err != nil {
		p.Log.Error("Error unmarshalling data")
		return nil
	}
        return p
}
```